### ISSUE TITLE
Special-case iterable of Time for Time initialization.

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#9882]
Currently, lists/tuples of `Time` instances are special cased in `_make_array` since the `np.array` call will give a warning for them in numpy >1.18 that being turned into object arrays is going to change. We should perhaps special-case iterable of `Time` so it never gets to this place. For more discussion, see https://github.com/astropy/astropy/pull/9719#discussion_r354242779

**Comment by SauravMaheshkar** [comment]
Should this issue still be open ?

**Comment by InCogNiTo124** [comment]
At a glance, I don't think it should

**Comment by pllim** [comment]
@mhvk , can you please confirm that this issue is still relevant or not? Thanks!

**Comment by mhvk** [comment]
Yes, it is still relevant. I updated the text on top with an example to show what the actual problem is.
